transactions/WFLY-17742_SupportTransactionsDuringGracefulShutdown

```diff
@@ -0,0 +1,146 @@
+*Proposed modifications:*
+* To address the shortcomings introduced with the FIFO order, a LIFO order should be used (discussed before)
+* With regards to the suspension logic of the EJB subsystem ([reference](https://github.com/wildfly/wildfly/blob/main/ejb3/src/main/java/org/jboss/as/ejb3/suspend/EJBSuspendHandlerService.java#L203)), the suspension handler of this subsystem should retain control (i.e. it should not return) as long as there are in-doubt transactions to be completed. This would guarantee data integrity, especially in situations when a negative timeout (i.e. indefinite waiting) is employed (e.g. in cloud environments). Further attention should be paid when the suspension logic is carried out asynchronously.
+
+===== Transactions
+
+*Current scenario.* The suspension logic developed in this subsystem is barely a call to the suspension hook of Narayana’s Recovery Coordinator, which only runs a single cycle to attempt the recovery of failed in-doubt transactions. As discussed previously, Narayana’s Recovery Coordinator does not necessarily recover all transactions before getting suspended as the “resume” action is supposed to happen (eventually). This will not necessarily be the case when WildFly is employed in cloud environments. Another important issue is that the current order (FIFO) used by the Suspend Controller during suspension results in a premature suspension of the Transactions subsystem. In other words, the recovery ability of this subsystem is suspended before dependent subsystems (e.g. EJB).
+*Proposed modifications:*
+* To address the shortcomings introduced with the FIFO order, a LIFO order should be used (discussed before)
+* The improvements to Narayana discussed above should be integrated in the Transactions subsystem
+  * In particular, the Transactions subsystem should become aware of how many in-doubt transactions there are in the Object Store and delay its suspension as long as all transactions are completed. In case (some of the) in-doubt transactions fail, the Transactions subsystem should wait as long as those transactions are resolved
+
+==== Note
+_Additional subsystems that depend on the Transactions subsystem should be added here, with details on how they should be handled._
+
+== Test Plan
+
+*wildfly-core.* As this proposal will not introduce new functionalities in wildfly-core, new testing is not needed.
+
+*Transactions and EJB subsystems.* Testing should be developed to make sure that in-doubt and in-flight transactions delay WildFly’s graceful shutdown. As a first step, we can test only WildFly on bare metal and then, if and when WildFly’s operator will be updated with modifications from this proposal, further testing might be developed in the cloud testing framework.
```

Comment:
I think the test plan should also include testing with multiple WildFly servers where transactions are propagated over EJB remoting and where a transaction is left in-doubt and where a transaction is inflight.

It would be interesting to configure a loop too to check that would work and not block: server A -> server B -> server A